source/Octopus.Tentacle/Util/OctopusPhysicalFileSystem.cs

```diff
@@ -257,7 +257,7 @@ public void EnsureDiskHasEnoughFreeSpace(string directoryPath, long requiredSpac
             if (IsUncPath(directoryPath))
                 return;
 
-            var driveInfo = new DriveInfo(Directory.GetDirectoryRoot(directoryPath));
+            var driveInfo = new DriveInfo(directoryPath);
```

Comment:
Yeah, I originally went with "only do this for non-windows", but digging into the code & docs, along with testing made me come to the conclusion it's all good. 

Agree with the feeling of nervousness though.

Reply:
Would a `Path.GetFullPath` get us out of trouble if someone inadvertently started throwing relative paths at this method?
All the code paths to get here look OK to me too, but it's a bad place to encounter an exception if a path isn't just right.

Reply:
I did contemplate doing a try/catch here, and falling back to the old approach if it didn't work, but I felt that it might be just hiding from the change, and leaving tech debt for the future.

One thing to note - the tests here all pass `Path.GetTempPath()`, so these tests are fully passing on all OS's. Does that give you more comfort? 

I can definitely add in a check to make sure the path is relative and convert it to absolute if need be.

Happy to add the try/catch if you would prefer us to be super safe?